### PR TITLE
Interface in dimension READY!

### PR DESCRIPTION
**Pull request by agorzawski**
Guys, 
The issue on Interfaces in dimensions is nearly done. Please look at the changes, I introduced two test classes, please start with those ones:

```
CoordinatesTest.java
CoordinatesOnInterfaceTest.java
```

**NOTE!!** tensext-cern-domain follow the changes with its CircularBuffer - I commit the changes once we close this issue.

### CODE REVIEW

File: src/java/org/tensorics/core/tensor/Position.java [hunk 1]
```diff
@@ -83,16 +83,16 @@ public static Position of(Object... coordinates) {
 		return createFrom(requireValidCoordinates(ImmutableMultiset.copyOf(coordinates)));
 	}
 
-	public ClassToInstanceMap<Object> getCoordinates() {
-		return coordinates;
+	public Set<Object> getCoordinates() {
+		return new HashSet<>(coordinates);
 	}
```
Comment (michi42) [inline src/java/org/tensorics/core/tensor/Position.java]: Do we really need getCoordinates() and coordinates(), as they now behave identically? I see the point of not breaking the API, but I think then we should deprecate one and remove it later...

File: src/java/org/tensorics/core/tensor/Position.java [hunk 2]
```diff
@@ -53,14 +52,15 @@
...
-	@SuppressWarnings("PMD.AvoidFieldNameMatchingMethodName")
-	private final ClassToInstanceMap<Object> coordinates;
+	private Set<?> coordinates;
 
 	private Position(Set<?> coordinates) {
-		this.coordinates = Coordinates.mapOf(coordinates);
-		if (this.coordinates.containsKey(Position.class)) {
-			throw new IllegalArgumentException("A position is contained in the collection of coordinates."
-					+ "This is most-probably a programming mistake and therefore not allowed.");
+		this.coordinates = new HashSet<>(coordinates);
```
Comment (michi42) [inline src/java/org/tensorics/core/tensor/Position.java]: I would suggest using an ImmutableSet here, so you can return it in coordinates() etc. without always making a copy.

**Comment by michi42** [comment]
Looks fine to me! Performance should be tested and could possibly be improved a bit by avoiding some copies in Position, but in principle this looks very nice!

One more thing ... perhaps we should somehow document or even deprecate Position.dimensionSet(), as it does not match the dimensions of the parent tensor anymore?

**Comment by agorzawski** [comment]
Good point Michi will document it. I will apply your suggestions as well!

**Pull request merged**